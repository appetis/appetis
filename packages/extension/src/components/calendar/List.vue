<template>
  <v-row class="fill-height">
    <v-col>
      <Loading />
      <v-sheet height="64">
        <v-toolbar flat>
          <v-btn fab small depressed @click="prev" class="mr-2 ml-1 py-5 px-5">
            <v-icon name="chevron-left" base-class="cv-icon"></v-icon>
          </v-btn>
          <v-btn
            fab
            small
            rounded
            depressed
            @click="next"
            class="mr-5 py-5 px-5"
          >
            <v-icon name="chevron-right" base-class="cv-icon"></v-icon>
          </v-btn>
          <v-btn
            rounded
            depressed
            class="mr-4 py-5 px-4"
            color="accent"
            @click="setToday"
          >
            <span>Today</span>
          </v-btn>
          <v-toolbar-title v-if="$refs.calendar">
            {{ $refs.calendar.title }}
          </v-toolbar-title>
          <v-spacer></v-spacer>

          <v-menu bottom right>
            <template v-slot:activator="{ on, attrs }">
              <v-btn
                rounded
                depressed
                class="py-5 px-4 mr-3"
                v-bind="attrs"
                v-on="on"
              >
                <span class="ml-2 mr-1">{{ typeToMe[me_type] }}</span>
                <v-icon name="chevron-down" base-class="cv-icon mx-1"></v-icon>
              </v-btn>
            </template>
            <v-list>
              <v-list-item @click="me_type = 'me'">
                <v-list-item-title>Me</v-list-item-title>
              </v-list-item>
              <v-list-item @click="me_type = 'all'">
                <v-list-item-title>All</v-list-item-title>
              </v-list-item>
            </v-list>
          </v-menu>

          <v-menu bottom right>
            <template v-slot:activator="{ on, attrs }">
              <v-btn
                rounded
                depressed
                class="py-5 px-4"
                v-bind="attrs"
                v-on="on"
              >
                <span class="ml-2 mr-1">{{ typeToLabel[type] }}</span>
                <v-icon name="chevron-down" base-class="cv-icon mx-1"></v-icon>
              </v-btn>
            </template>
            <v-list>
              <v-list-item @click="type = 'month'">
                <v-list-item-title>Month</v-list-item-title>
              </v-list-item>
              <v-list-item @click="type = 'week'">
                <v-list-item-title>Week</v-list-item-title>
              </v-list-item>
              <v-list-item @click="type = 'day'">
                <v-list-item-title>Day</v-list-item-title>
              </v-list-item>
            </v-list>
          </v-menu>
        </v-toolbar>
      </v-sheet>


      <v-sheet height="650">

        <v-calendar
          ref="calendar"
          v-model="focus"
          color="primary"
          :events="events"
          :event-color="getEventColor"
          :type="type"
          @click:event="showEvent"
          @click:more="viewDay"
          @click:date="viewDay"
          @change="updateRange"
        ></v-calendar>

        <v-menu
          v-model="selectedOpen"
          :close-on-content-click="false"
          :activator="selectedElement"
          offset-x
        >
          <v-card color="grey lighten-4" min-width="350px" flat>
            <v-toolbar :color="selectedEvent.color" dark>
              <v-btn icon>
                <v-icon name="edit-2" base-class="v-icon"></v-icon>
              </v-btn>
              <v-toolbar-title v-html="selectedEvent.name"></v-toolbar-title>
              <v-spacer></v-spacer>
              <v-btn icon>
                <v-icon name="heart" base-class="v-icon"></v-icon>
              </v-btn>
              <v-btn icon>
                <v-icon name="more-vertical" base-class="v-icon"></v-icon>
              </v-btn>
            </v-toolbar>
            <v-card-text>
              <span v-html="selectedEvent.details"></span>
            </v-card-text>
            <v-card-actions>
              <v-btn text color="secondary" @click="selectedOpen = false">
                Cancel
              </v-btn>
            </v-card-actions>
          </v-card>
        </v-menu>
      </v-sheet>
    </v-col>
  </v-row>
</template>
<script>
import Loading from "../loading/Loading";
export default {
  components: {
    Loading
  },
  data: () => ({
    focus: '',
    me_type: 'me',
    typeToMe: {
      me: 'Me',
      all: 'All',
    },
    type: 'month',
    typeToLabel: {
      month: 'Month',
      week: 'Week',
      day: 'Day',
    },
    selectedEvent: {},
    selectedElement: null,
    selectedOpen: false,
    events: [],
    colors: [
      'ca-white',
      'ca-green',
      'ca-blue',
      'ca-purple',
      'ca-orange',
      // 'orange',
      // 'grey darken-1',
    ],
    names: [
      'Meeting',
      'Holiday',
      'PTO',
      'Travel',
      'Event',
      'Birthday',
      'Conference',
      'Party',
    ],
  }),
  mounted() {
    this.$refs.calendar.checkChange()
  },
  created() {
    this.initialize()
  },
  methods: {
    initialize() {
      console.log("calendar page init")
    },
    viewDay({ date }) {
      this.focus = date
      this.type = 'day'
    },
    getEventColor(event) {
      return event.color
    },
    setToday() {
      this.focus = ''
    },
    prev() {
      this.$refs.calendar.prev()
    },
    next() {
      this.$refs.calendar.next()
    },
    showEvent({ nativeEvent, event }) {
      console.log(event)
      const open = () => {
        this.selectedEvent = event
        this.selectedElement = nativeEvent.target
        requestAnimationFrame(() =>
          requestAnimationFrame(() => (this.selectedOpen = true))
        )
      }

      if (this.selectedOpen) {
        this.selectedOpen = false
        requestAnimationFrame(() => requestAnimationFrame(() => open()))
      } else {
        open()
      }

      nativeEvent.stopPropagation()
    },
    updateRange({ start, end }) {
      const events = []

      const min = new Date(`${start.date}T00:00:00`)
      const max = new Date(`${end.date}T23:59:59`)
      const days = (max.getTime() - min.getTime()) / 86400000
      const eventCount = this.rnd(days, days + 20)

      for (let i = 0; i < eventCount; i++) {
        const allDay = this.rnd(0, 3) === 0
        const firstTimestamp = this.rnd(min.getTime(), max.getTime())
        const first = new Date(firstTimestamp - (firstTimestamp % 900000))
        const secondTimestamp = this.rnd(2, allDay ? 288 : 8) * 900000
        const second = new Date(first.getTime() + secondTimestamp)

        events.push({
          name: this.names[this.rnd(0, this.names.length - 1)],
          start: first,
          end: second,
          color: this.colors[this.rnd(0, this.colors.length - 1)],
          timed: !allDay,
        })
      }
      console.log(events)

      this.events = events
      this.$store.dispatch('setLoading', false)
    },
    rnd(a, b) {
      return Math.floor((b - a + 1) * Math.random()) + a
    },
  },
}
</script>

<style scoped>
.cv-icon {
  width: 18px;
  color: #151515;
}
</style>
