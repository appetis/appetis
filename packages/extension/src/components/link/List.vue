<template>
  <div>
    <Loading :isLoading="isLoading" />
    <p>Link List</p>
  </div>
</template>

<script>
import Loading from '../loading/Loading'
export default {
  name: 'Link',
  // mounted() {
  //   browser.runtime.sendMessage({});
  // },
  data() {
    return {
      isLoading: true,
    }
  },
  computed: {},
  components: {
    Loading,
  },
  created() {
    this.initialize()
  },
  methods: {
    initialize() {
      console.log('Link page init')
      window.setTimeout(() => {
        this.$store.dispatch('setLoading', false)
      }, 3000)
    },
  },
}
</script>

<style scoped>
p {
  font-size: 20px;
}
</style>
