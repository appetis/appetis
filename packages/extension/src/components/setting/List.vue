<template>
  <div>
    <Loading :isLoading="isLoading" />
    <p>Setting List</p>

    <div class="ml-4 font-bold cursor-pointer" @click="logout()">Logout</div>
  </div>
</template>

<script>
import Loading from '../loading/Loading'
export default {
  name: 'Setting',
  // mounted() {
  //   browser.runtime.sendMessage({});
  // },
  components: {
    Loading,
  },
  data() {
    return {
      isLoading: false,
    }
  },
  computed: {},
  created() {
    this.initialize()
  },
  methods: {
    initialize() {
      console.log('Setting page init')

      this.$store.dispatch('setLoading', false)
    },
    logout() {
      this.$root.$emit('getStatusStop')
      this.$store.dispatch('setLoading', true)
      this.$store
        .dispatch('logout')
        .then((response) => {
          console.log('logout --->', response)
          this.$root.$emit('resetPage')
          this.$store.dispatch('setLoading', false)
        })
        .catch((error) => {
          this.showAuthenticationFailMessage = error.response.data.message
          this.isLoading = false
          this.$store.dispatch('setLoading', false)
        })
    },
  },
}
</script>

<style scoped>
p {
  font-size: 20px;
}
</style>
