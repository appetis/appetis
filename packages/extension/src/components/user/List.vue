<template>
  <div>
    <Loading :isLoading="isLoading" />
    <v-data-table
      :headers="headers"
      :items="users"
      sort-by="department"
      :single-select="singleSelect"
      item-key="name"
      loading
      loading-text="Loading... Please wait"
    >
      <v-progress-linear
        v-show="false"
        :size="70"
        :width="7"
        slot="progress"
        color="#385859"
        indeterminate
      ></v-progress-linear>
      <template v-slot:item.Profile.name="{ item }">
        <div class="sm-profile-photo" v-bind:class="item.status_class">
          <div class="inline-block">
            <v-avatar
              color="teal"
              size="24"
              class="header-no-avatar pt-1.5 -mt-2"
              v-show="!item.Profile.avatar"
            >
              <span class="white--text text-xs mt-2">HW</span>
            </v-avatar>
            <img
              :src="item.Profile.avatar"
              class="profile-avatar"
              v-show="item.Profile.avatar"
            />
          </div>

          <div class="inline-block ml-3">
            {{ item.Profile.name }}
          </div>
        </div>
      </template>
      <template v-slot:item.Profile.status="{ item }">
        <v-chip :color="getProfileClass(item.Profile.status)" dark> </v-chip>
        {{ item.Profile.statusName }}
      </template>
      <template v-slot:top>
        <v-toolbar flat>
          <v-toolbar-title>Members</v-toolbar-title>
          <span class="sub-title"> {{ users.length }} members </span>
          <v-spacer></v-spacer>
          <v-dialog v-model="dialog" max-width="500px">
            <template v-slot:activator="{ on, attrs }">
              <div
                class="btn-add-member"
                v-bind="attrs"
                v-on="on"
                v-show="isReady"
              >
                <v-icon name="user-plus" base-class="icon"></v-icon>
                Add Member
              </div>
            </template>
            <v-card id="modal-add-member">
              <v-card-title>
                <span class="text-h5">{{ formTitle }}</span>
              </v-card-title>

              <v-card-text>
                <v-container>
                  <v-row>
                    <v-col cols="12" sm="12" md="12">
                      <v-text-field
                        v-model="editedItem.name"
                        label="Name"
                      ></v-text-field>
                    </v-col>
                    <v-col cols="12" sm="6" md="6">
                      <v-text-field
                        v-model="editedItem.department"
                        label="Department"
                      ></v-text-field>
                    </v-col>
                    <v-col cols="12" sm="6" md="6">
                      <v-text-field
                        v-model="editedItem.position"
                        label="Position"
                      ></v-text-field>
                    </v-col>
                    <v-col cols="12" sm="6" md="6">
                      <v-text-field
                        v-model="editedItem.email"
                        label="Email"
                      ></v-text-field>
                    </v-col>
                    <v-col cols="12" sm="6" md="6">
                      <v-text-field
                        v-model="editedItem.phone"
                        label="Phone"
                      ></v-text-field>
                    </v-col>
                    <v-col cols="12" sm="6" md="6">
                      <v-text-field
                        v-model="editedItem.status"
                        label="Status"
                      ></v-text-field>
                    </v-col>
                  </v-row>
                </v-container>
              </v-card-text>

              <v-card-actions>
                <v-spacer></v-spacer>
                <v-btn color="blue darken-1" text @click="close">
                  Cancel
                </v-btn>
                <v-btn
                  color="blue darken-1"
                  id="btn-save-member"
                  text
                  @click="save"
                >
                  Save
                </v-btn>
              </v-card-actions>
            </v-card>
          </v-dialog>
          <v-dialog v-model="dialogDelete" max-width="500px">
            <v-card>
              <v-card-title class="text-h5"
                >Are you sure you want to delete this item?</v-card-title
              >
              <v-card-actions>
                <v-spacer></v-spacer>
                <v-btn color="blue darken-1" text @click="closeDelete"
                  >Cancel</v-btn
                >
                <v-btn color="blue darken-1" text @click="deleteItemConfirm"
                  >OK</v-btn
                >
                <v-spacer></v-spacer>
              </v-card-actions>
            </v-card>
          </v-dialog>
          <v-dialog v-model="dialogProfile" max-width="500px">
            <Profile
              v-show="dialogProfile"
              @close="closeProfile"
              :profile="editedItem"
              id="modal-profile"
              ref="profile"
            />
          </v-dialog>
        </v-toolbar>
      </template>

      <template v-slot:item.actions="{ item }">
        <div class="action-box" @click="openProfile(item)">
          <v-icon name="user" base-class="icon-12"></v-icon>
        </div>

        <!-- For version 2
        <div class="action-box" @click="editItem(item)">
          <v-icon name="calendar" base-class="icon-12"></v-icon>
        </div>
        <div class="action-box" @click="deleteItem(item)">
          <v-icon name="mail" base-class="icon-12"></v-icon>
        </div>
        <div class="action-box" @click="deleteItem(item)">
          <v-icon name="message-square" base-class="icon-12"></v-icon>
        </div>
        -->
      </template>

      <template v-slot:no-data>
        <v-btn color="primary" @click="initialize"> Reset </v-btn>
      </template>
    </v-data-table>
  </div>
</template>

<script>
import { mapGetters } from 'vuex'
import TeamService from '@/services/TeamService'
import Profile from './Profile'
import Loading from '../loading/Loading'
//import { Circle2 } from 'vue-loading-spinner'

export default {
  name: 'User',
  components: {
    Profile,
    Loading,
  },
  data() {
    return {
      isLoading: true,
      isReady: false,
      singleSelect: false,
      selected: [],
      dialog: false,
      dialogDelete: false,
      dialogProfile: false,
      headers: [
        {
          text: 'Name',
          align: 'start',
          sortable: false,
          value: 'Profile.name',
        },
        { text: 'Department', value: 'Profile.department' },
        { text: 'Position', value: 'Profile.position' },
        { text: 'Email', value: 'email' },
        { text: 'Phone', value: 'Profile.phone' },
        { text: 'Status', value: 'Profile.status' },
        //{ text: 'Action', value: 'actions', sortable: false },
      ],
      users: [],
      editedIndex: -1,
      editedItem: {
        name: '1',
        department: '2',
        position: '3',
        email: '4',
        phone: '5',
        status: '6',
      },
      defaultItem: {
        name: '',
        department: '',
        position: '',
        email: '',
        phone: '',
        status: '',
      },
      totalCount: 0,
      selectedAvatar: null,
      selectedProfile: null,
    }
  },
  computed: {
    formTitle() {
      return this.editedIndex === -1 ? 'Add User' : 'Edit User'
    },
    ...mapGetters(['getTeams', 'getTeamsCount']),
  },
  watch: {
    dialog(val) {
      val || this.close()
    },
    dialogDelete(val) {
      val || this.closeDelete()
    },
    dialogProfile(val) {
      val || this.closeProfile()
    },
  },

  created() {
    this.initialize()
  },

  methods: {
    async initialize() {
      //console.log("======> initialize", this.$store.state.user.teams.length, this.$store.state.user.teams[0].id)
      //console.log('===== init List ', this.$store.state, this.getTeams, this.getTeamsCount)

      if (this.getTeamsCount) {
        const team_id = this.getTeams[0].id
        await TeamService.getTeams(team_id)
          .then((response) => {
            //console.log(response.data.team)
            this.users = response.data.team.members
            this.$store.dispatch('setLoading', false)
            this.isReady = true
          })
          .catch((error) => {
            throw error
          })
      }
      /*this.users = [
        {
          name: 'Frozen Yogurt',
          position: 'Project Manager',
          email: 'random@email.com',
          phone: '(647) 999-1234',
          status: 'OF',
          status_name: 'In office',
        },
        {
          name: 'Ice cream sandwich',
          position: 'UX/UI Designer',
          email: 'test1@email.com',
          phone: '(647) 999-5678',
          status: 'OF',
          status_name: 'In office',
        },
        {
          name: 'Eclair',
          position: 'Full-Stack Developer',
          email: 'test2@email.com',
          phone: '(647) 999-0000',
          status: 'WH',
          status_name: 'Work from home',
        },
        {
          name: 'Cupcake',
          position: 'Full-Stack Developer',
          email: 'test2@email.com',
          phone: '(647) 999-0000',
          status: 'VE',
          status_name: 'On vacation',
        },
        {
          name: 'Gingerbread',
          position: 'Full-Stack Developer',
          email: 'test2@email.com',
          phone: '(647) 999-0000',
          status: 'VE',
          status_name: 'On vacation',
        },
        {
          name: 'Jelly bean',
          position: 'Project Manager',
          email: 'random@email.com',
          phone: '(647) 999-1234',
          status: 'OF',
          status_name: 'In office',
        },
        {
          name: 'Lollipop',
          position: 'Project Manager',
          email: 'random@email.com',
          phone: '(647) 999-1234',
          status: 'WH',
          status_name: 'Work from home',
        },
        {
          name: 'Honeycomb',
          position: 'Full-Stack Developer',
          email: 'test2@email.com',
          phone: '(647) 999-0000',
          status: 'WH',
          status_name: 'Work from home',
        },
        {
          name: 'Donut',
          position: 'Full-Stack Developer',
          email: 'test2@email.com',
          phone: '(647) 999-0000',
          status: 'WH',
          status_name: 'Work from home',
        },
        {
          name: 'KitKat',
          position: 'Full-Stack Developer',
          email: 'test2@email.com',
          phone: '(647) 999-0000',
          status: 'WH',
          status_name: 'Work from home',
        },
      ]*/
    },

    editItem(user) {
      this.editedIndex = this.users.indexOf(user)
      this.editedItem = Object.assign({}, user)
      this.dialog = true
    },
    deleteItem(user) {
      this.editedIndex = this.users.indexOf(user)
      this.editedItem = Object.assign({}, user)
      this.dialogDelete = true
    },
    openProfile(user) {
      this.editedIndex = this.users.indexOf(user)
      this.editedItem = Object.assign({}, user)
      this.selectedProfile = user
      this.selectedProfile.Profile.status_class =
        'profile-' + this.getProfileClass(user.Profile.status)
      this.dialogProfile = true
    },

    messageUser(user) {
      this.editedIndex = this.users.indexOf(user)
      this.editedItem = Object.assign({}, user)
      this.dialogDelete = true
    },

    deleteItemConfirm() {
      this.users.splice(this.editedIndex, 1)
      this.closeDelete()
    },

    close() {
      this.dialog = false
      this.$nextTick(() => {
        this.editedItem = Object.assign({}, this.defaultItem)
        this.editedIndex = -1
      })
    },

    closeDelete() {
      this.dialogDelete = false
      this.$nextTick(() => {
        this.editedItem = Object.assign({}, this.defaultItem)
        this.editedIndex = -1
      })
    },

    closeProfile() {
      this.dialogProfile = false
      this.$nextTick(() => {
        // this.editedItem = Object.assign({}, this.defaultItem)
        // this.editedIndex = -1
      })
    },

    save() {
      if (this.editedIndex > -1) {
        Object.assign(this.users[this.editedIndex], this.editedItem)
      } else {
        const profile = this.editedItem
        const user = {
          id: 0,
          email: this.editedItem.email,
          Profile: profile,
        }
        this.users.push(user)
      }
      this.close()
    },
    getProfileClass(status) {
      let status_color = ''
      switch (status) {
        case 'VA':
          status_color = 'vacation'
          break
        case 'OF':
          status_color = 'online'
          break
        default:
          status_color = 'other'
      }
      return status_color
    },
  },
}
</script>

<style scoped>
.v-toolbar__title {
  @apply ml-2;
}
</style>
