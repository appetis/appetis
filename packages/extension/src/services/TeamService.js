import apiClient from './apiClient'

export default {
  getTeams(teamId) {
    return apiClient.get(`/api/teams/${teamId}/members`)
  },
}
