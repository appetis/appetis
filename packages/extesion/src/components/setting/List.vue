<template>
  <div>
    <p>Setting List</p>
  </div>
</template>

<script>
export default {
  name: 'Setting',
  // mounted() {
  //   browser.runtime.sendMessage({});
  // },
  computed: {},
}
</script>

<style scoped>
p {
  font-size: 20px;
}
</style>
