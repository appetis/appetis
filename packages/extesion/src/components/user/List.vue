<template>
  <div>
    <p>User List</p>
  </div>
</template>

<script>
export default {
  name: 'User',
  // mounted() {
  //   browser.runtime.sendMessage({});
  // },
  computed: {
    defaultText() {
      return browser.i18n.getMessage('extName')
    },
  },
}
</script>

<style scoped>
p {
  font-size: 20px;
}
</style>
