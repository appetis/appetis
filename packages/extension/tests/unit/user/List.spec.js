import List from '@/components/user/List'
import Vuetify from 'vuetify'
import { createLocalVue, mount } from '@vue/test-utils'
import flushPromises from 'flush-promises'
import TeamService from '@/services/TeamService'
import Vuex from 'vuex'
import { mockTeams } from '../../data'

jest.mock('@/services/TeamService')
const localVue = createLocalVue()

localVue.use(Vuex)

let getters
let actions
let store
let vuetify

beforeEach(() => {
  jest.clearAllMocks()

  getters = {
    getTeams: () => [{ id: 1 }],
    getTeamsCount: () => 1,
  }
  actions = {
    setLoading: jest.fn(),
  }
  store = new Vuex.Store({
    getters,
    actions,
  })

  vuetify = new Vuetify()
})
const factory = async () => {
  TeamService.getTeams.mockResolvedValueOnce({ data: mockTeams() })
  const wrapper = mount(List, { store, localVue, vuetify })
  await flushPromises()
  return wrapper
}
describe('user/List.vue', () => {
  document.body.setAttribute('data-app', true)

  it('Calls getTeams and displays teams', async () => {
    //getTeam.mockResolvedValueOnce({ "user": mockTeams() })
    const wrapper = await factory()
    await wrapper.vm.$nextTick()
    expect(TeamService.getTeams).toHaveBeenCalledTimes(1)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('Click add Member button and click save button', async () => {
    const wrapper = await factory()

    //const event = jest.fn()

    // wrapper.vm.$on('action-btn:clicked', event)
    // expect(event).toHaveBeenCalledTimes(0)

    // await wrapper.find('.btn-add-member').trigger('click')
    // expect(wrapper.find('.v-card').isVisible()).toBe(true)
    // expect(event).toHaveBeenCalledTimes(1)

    //const wrapper = mount(List, { store, localVue, vuetify })

    await wrapper.find('.btn-add-member').trigger('click')
    expect(wrapper.find('#modal-add-member').isVisible()).toBe(true)
    expect(wrapper.find('#btn-save-member').isVisible()).toBe(true)

    await wrapper.find('#btn-save-member').trigger('click')
    expect(
      wrapper.findAll('.v-data-table__wrapper table tbody tr').length
    ).toBe(mockTeams().team.members.length + 1)
  })
})
