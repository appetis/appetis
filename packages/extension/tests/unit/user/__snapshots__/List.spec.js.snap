// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`user/List.vue Calls getTeams and displays teams 1`] = `
<div>
  <div>
    <div role="progressbar" aria-valuemin="0" aria-valuemax="100" class="v-progress-linear v-progress-linear--visible theme--light" style="height: 4px; display: none;" slot="progress">
      <div class="v-progress-linear__background" style="opacity: 0.3; left: 0%; width: 100%; background-color: rgb(77, 165, 170); border-color: #4da5aa;"></div>
      <div class="v-progress-linear__buffer"></div>
      <transition-stub name="fade-transition">
        <div class="v-progress-linear__indeterminate v-progress-linear__indeterminate--active">
          <div class="v-progress-linear__indeterminate long" style="background-color: rgb(77, 165, 170); border-color: #4da5aa;"></div>
          <div class="v-progress-linear__indeterminate short" style="background-color: rgb(77, 165, 170); border-color: #4da5aa;"></div>
        </div>
      </transition-stub>
    </div>
  </div>
  <div class="v-data-table v-data-table--has-top v-data-table--has-bottom theme--light">
    <header class="v-sheet theme--light v-toolbar v-toolbar--flat" style="height: 64px;">
      <div class="v-toolbar__content" style="height: 64px;">
        <div class="v-toolbar__title">Members</div> <span class="sub-title"> 2 members </span>
        <div class="spacer"></div>
        <div role="dialog" class="v-dialog__container">
          <!---->
        </div>
        <div role="button" aria-haspopup="true" aria-expanded="false" class="btn-add-member" style=""><svg viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="icon icon-user-plus">
            <path d="M16 21v-2a4 4 0 0 0-4-4H5a4 4 0 0 0-4 4v2"></path>
            <circle cx="8.5" cy="7" r="4"></circle>
            <line x1="20" x2="20" y1="8" y2="14"></line>
            <line x1="23" x2="17" y1="11" y2="11"></line>
          </svg>
          Add Member
        </div>
        <div role="dialog" class="v-dialog__container">
          <!---->
        </div>
        <div role="dialog" class="v-dialog__container">
          <!---->
        </div>
      </div>
    </header>
    <div class="v-data-table__wrapper">
      <table>
        <colgroup>
          <col class="">
          <col class="">
          <col class="">
          <col class="">
          <col class="">
          <col class="">
          <col class="">
        </colgroup>
        <thead class="v-data-table-header">
          <tr>
            <th role="columnheader" scope="col" aria-label="Name" class="text-start"><span>Name</span></th>
            <th role="columnheader" scope="col" aria-label="Department: Not sorted. Activate to sort ascending." aria-sort="none" class="text-start sortable"><span>Department</span><i aria-hidden="true" class="v-icon notranslate v-data-table-header__icon mdi mdi-arrow-up theme--light" style="font-size: 18px;"></i></th>
            <th role="columnheader" scope="col" aria-label="Position: Not sorted. Activate to sort ascending." aria-sort="none" class="text-start sortable"><span>Position</span><i aria-hidden="true" class="v-icon notranslate v-data-table-header__icon mdi mdi-arrow-up theme--light" style="font-size: 18px;"></i></th>
            <th role="columnheader" scope="col" aria-label="Email: Not sorted. Activate to sort ascending." aria-sort="none" class="text-start sortable"><span>Email</span><i aria-hidden="true" class="v-icon notranslate v-data-table-header__icon mdi mdi-arrow-up theme--light" style="font-size: 18px;"></i></th>
            <th role="columnheader" scope="col" aria-label="Phone: Not sorted. Activate to sort ascending." aria-sort="none" class="text-start sortable"><span>Phone</span><i aria-hidden="true" class="v-icon notranslate v-data-table-header__icon mdi mdi-arrow-up theme--light" style="font-size: 18px;"></i></th>
            <th role="columnheader" scope="col" aria-label="Status: Not sorted. Activate to sort ascending." aria-sort="none" class="text-start sortable"><span>Status</span><i aria-hidden="true" class="v-icon notranslate v-data-table-header__icon mdi mdi-arrow-up theme--light" style="font-size: 18px;"></i></th>
            <th role="columnheader" scope="col" aria-label="Action" class="text-start"><span>Action</span></th>
          </tr>
        </thead>
        <thead>
          <tr class="v-data-table__progress">
            <th colspan="7" class="column">
              <div role="progressbar" aria-valuemin="0" aria-valuemax="100" class="v-progress-linear v-progress-linear--visible theme--light" style="height: 4px; display: none;" size="70" width="7">
                <div class="v-progress-linear__background" style="opacity: 0.3; left: 0%; width: 100%; background-color: rgb(56, 88, 89); border-color: #385859;"></div>
                <div class="v-progress-linear__buffer"></div>
                <transition-stub name="fade-transition">
                  <div class="v-progress-linear__indeterminate v-progress-linear__indeterminate--active">
                    <div class="v-progress-linear__indeterminate long" style="background-color: rgb(56, 88, 89); border-color: #385859;"></div>
                    <div class="v-progress-linear__indeterminate short" style="background-color: rgb(56, 88, 89); border-color: #385859;"></div>
                  </div>
                </transition-stub>
              </div>
            </th>
          </tr>
        </thead>
        <tbody>
          <tr class="">
            <td class="text-start">Wonho Lee</td>
            <td class="text-start">IT</td>
            <td class="text-start">Developer</td>
            <td class="text-start">wonho@workly.page</td>
            <td class="text-start">111-111-1111</td>
            <td class="text-start"><span class="v-chip theme--dark v-size--default online"><span class="v-chip__content"></span></span>
              In office
            </td>
            <td class="text-start">
              <div class="action-box"><svg viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="icon-12 icon-user">
                  <path d="M20 21v-2a4 4 0 0 0-4-4H8a4 4 0 0 0-4 4v2"></path>
                  <circle cx="12" cy="7" r="4"></circle>
                </svg></div>
            </td>
          </tr>
          <tr class="">
            <td class="text-start">Hyewon Kim</td>
            <td class="text-start">IT</td>
            <td class="text-start">Developer</td>
            <td class="text-start">hyewon@workly.page</td>
            <td class="text-start">222-222-2222</td>
            <td class="text-start"><span class="v-chip theme--dark v-size--default other"><span class="v-chip__content"></span></span>
              Work from home
            </td>
            <td class="text-start">
              <div class="action-box"><svg viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="icon-12 icon-user">
                  <path d="M20 21v-2a4 4 0 0 0-4-4H8a4 4 0 0 0-4 4v2"></path>
                  <circle cx="12" cy="7" r="4"></circle>
                </svg></div>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
    <div class="v-data-footer">
      <div class="v-data-footer__select">Rows per page:<div class="v-input v-input--hide-details v-input--is-label-active v-input--is-dirty theme--light v-text-field v-select">
          <div class="v-input__control">
            <div role="button" aria-haspopup="listbox" aria-expanded="false" aria-owns="list-24" class="v-input__slot">
              <div class="v-select__slot">
                <div class="v-select__selections">
                  <div class="v-select__selection v-select__selection--comma">10</div><input aria-label="Rows per page:" id="input-24" readonly="readonly" type="text" aria-readonly="false" autocomplete="off">
                </div>
                <div class="v-input__append-inner">
                  <div class="v-input__icon v-input__icon--append"><i aria-hidden="true" class="v-icon notranslate mdi mdi-menu-down theme--light"></i></div>
                </div><input type="hidden" value="10">
              </div>
              <div class="v-menu">
                <!---->
              </div>
            </div>
          </div>
        </div>
      </div>
      <div class="v-data-footer__pagination">1-2 of 2</div>
      <div class="v-data-footer__icons-before"><button type="button" disabled="disabled" class="v-btn v-btn--disabled v-btn--icon v-btn--round v-btn--text theme--light v-size--default" aria-label="Previous page"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-chevron-left theme--light"></i></span></button></div>
      <div class="v-data-footer__icons-after"><button type="button" disabled="disabled" class="v-btn v-btn--disabled v-btn--icon v-btn--round v-btn--text theme--light v-size--default" aria-label="Next page"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-chevron-right theme--light"></i></span></button></div>
    </div>
  </div>
</div>
`;
